extensions/resteasy-classic/resteasy-client/deployment/src/test/java/io/quarkus/restclient/exception/SubResourceTest.java

```diff
@@ -0,0 +1,165 @@
+package io.quarkus.restclient.exception;
+
+import static jakarta.ws.rs.core.MediaType.TEXT_PLAIN;
+import static jakarta.ws.rs.core.Response.Status.OK;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.rest.client.annotation.ClientHeaderParam;
+import org.eclipse.microprofile.rest.client.annotation.RegisterProvider;
+import org.eclipse.microprofile.rest.client.ext.ResponseExceptionMapper;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+/**
+ * Tests client sub-resources
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+public class SubResourceTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(SubResourceTest.class, ClientRootResource.class, ClientSubResource.class,
+                            ServerResource.class, TestExceptionMapper.class, TestException.class)
+                    .addAsResource(new StringAsset(ClientRootResource.class.getName() + "/mp-rest/url=${test.url}\n"),
+                            "application.properties"));
+
+    @RestClient
+    ClientRootResource clientRootResource;
+
+    /**
+     * Creates a REST client with an attached exception mapper. The exception mapper will throw a {@link TestException}.
+     * This test invokes a call to the root resource.
+     *
+     */
+    @Test
+    public void rootResourceExceptionMapper() {
+        try (final Response ignored = clientRootResource.fromRoot()) {
```

Comment:
This might be minor, but `final` is redundant for resources like this. Some projects check-styles fail on this.

Reply:
just removed it